src/riak_ql_ddl_compiler.erl

```diff
@@ -299,6 +301,16 @@ build_get_ddl_fn(DDL, LineNo, []) ->
     Fn = flat_format("get_ddl() -> ~p.", [DDL]),
     {?Q(Fn), LineNo + 1}.
 
+build_delete_key_fn(DDL, LineNo, []) ->
+    Fn = flat_format("get_delete_key(W) ->"
+                     "LocalKey = ~p, "
+                     "case riak_ql_ddl_util:is_valid_delete_where_clause(W) of "
+                     "true -> riak_ql_ddl_util:make_delete_key(LocalKey, W); "
+                     "{error, Errors} -> {false, Errors} "
```

Comment:
why not {error, Errors} ? The corresponding case will be comparing {ok, _} | {false, _}, but {ok, _} | {error, _} is idiomatic _afaik_.